wandb/sdk/wandb_settings.py

```diff
@@ -426,6 +426,10 @@ def _kaggle(self) -> bool:
     def _windows(self) -> bool:
         return platform.system() == "Windows"
 
+    @property
+    def _sync_tensorboard(self) -> bool:
+        return len(wandb.patched["tensorboard"]) > 0
+
```

Comment:
putting a dynamic property here is dangerous. 
For example.

We convert this settings object into a static object when we create the internal processs.  So anything in wandb/sdk/internal is going to point to a snapshot of this value which might not be correct, or if it is, just by luck

Reply:
Okay, I see. I think this might be a little tricky then. I'm going to try and find an alternative way to communicate this to handler.py

Reply:
I was envisioning that the backend wouldn't care if the history came from one place or the other and we could consolidate step increases here for both wandb.log() and sync_tensorboard.

Reply:
The backend needs to know if we're in sync_tensorboard mode because that's the only mode that should ignore the `step` values from user calls to `wandb.log`...